vip-jetpack/connection-pilot/class-jetpack-connection-pilot.php

```diff
@@ -255,7 +260,12 @@ public static function should_run_connection_pilot() {
 	 * @param $error \WP_Error Optional error thrown by the connection check
 	 * @return bool True if a reconnect should be attempted
 	 */
-	public static function should_attempt_reconnection( $error = null ) {
+	public static function should_attempt_reconnection( \WP_Error $error = null ): bool {
+		// Do not try to reconnect anything else other than production sites
+		if ( 'production' !== VIP_GO_APP_ENVIRONMENT ) {
+			return false;
+		}
```

Comment:
Forcing non-prod environments to avoid reconnection. Is it ok that we introduce this conditional here, or we prefer to rely solely in the global variable properly set up?